### ISSUE TITLE
Migration to pillow-7.0.0 and python-3?

### ISSUE THREAD

**Issue by Massimo-B** [jepler/cropgui#68]
On Gentoo the very last development version conflicts about python pillow library:
```
WARNING: One or more updates/rebuilds have been skipped due to a dependency conflict:

dev-python/pillow:0

  (dev-python/pillow-7.0.0:0/0::gentoo, ebuild scheduled for merge) USE="jpeg lcms tiff truetype zlib -doc -examples -imagequant -jpeg2k -test -tk -webp" ABI_X86="(64)" PYTHON_TARGETS="python3_6 -python3_7 -python3_8" conflicts with
    dev-python/pillow[python_targets_python2_7(-),-python_single_target_python2_7(-)] required by (media-gfx/cropgui-9999:0/0::gentoo, installed) USE="" ABI_X86="(64)" PYTHON_TARGETS="python2_7"
```
As far as I understand, the cropgui depends on pillow with python2_7 support where pillow-6.2.2 is the last matching. The new pillow-7.0.0 only supports python3_6, 3_7 and 3_8.
cropgui itself on Gentoo is depending on python2_7 and thus being masked due to global distribution migration to python3.

Are there plans to migrate to python3?

**Comment by jepler** [comment]
Pull requests welcome. I thought I had previously merged what was required for compatible with

**Comment by waptaff** [comment]
@Massimo-B I suggest you check for the updated ebuild on Gentoo's bugtracker.  I can build cropgui just fine with it. https://bugs.gentoo.org/708098#c2

Note that for the time being you may have to user-patch if an issue with cropped image not matching what you selected hits you.  If so, just drop the file attached to this comment into `/etc/portage/patches/media-gfx/cropgui/integer-scaling.patch`, re-emerge and you should be fine; that fix was merged after the 0.4 release.

[integer-scaling.patch.txt](https://github.com/jepler/cropgui/files/4313714/integer-scaling.patch.txt)

**Comment by Massimo-B** [comment]
Thanks, compiled and working. I'm going to push if that proposed ebuild will be released.
